### PR TITLE
First-class object support

### PR DESCRIPTION
**Pull request by mxstbr**
With this patch you can now do

```JS
const Component = styled.div({
  color: 'blue'
})
```

This is in response to https://mobile.twitter.com/kentcdodds/status/994254700651339776, /cc @kentcdodds

### CODE REVIEW

**Comment by Gabri3l** [comment]
Now my life is complete. Thank you!

File: src/constructors/css.js [hunk 1]
```diff
@@ -4,6 +4,11 @@ import flatten from '../utils/flatten'
 import type { Interpolation, RuleSet } from '../types'
 
 export default (
-  strings: Array<string>,
+  strings: Array<string> | Object,
```
Comment (probablyup) [inline src/constructors/css.js]: We should probably rename this property. Perhaps "styles"?

Comment (mxstbr) [reply src/constructors/css.js]: Yeah that makes sense, incoming.

**Comment by ghengeveld** [comment]
Can you also do this? 

```js
const Component = styled.div({
  color: 'blue'
}, {
  width: 100
}, props => ({
  background: props.bg
}))
```

A contrived example but the idea is to have multiple style objects stacked on top of each other, done even being the callback style.

**Pull request merged**